### ISSUE TITLE
fix error special characters in name json

### ISSUE THREAD

**Issue by Rot4tion** [barbarbar338/bh-open-api#2]
example:
★Kurisu Makise★ <3 = âKurisu Makiseâ <3
Pajé 🪐 = PajÃ© ðª

Thank you for reading <3.

**Comment by barbarbar338** [comment]
I'll check ASAP

**Comment by Rot4tion** [comment]
> I'll check ASAP

I see you still haven't fixed it.
here's how to fix it

https://github.com/barbarbar338/bh-open-api/blob/main/src/libs/BHAPI.ts -> makeAPIRequest
test: https://brawlhalla.fly.dev/v1/stats/id?brawlhalla_id=31682773
```
const response = await axios.get(urlFormat, {
        transformResponse: (data) => {
          var jsonClean = decodeURIComponent(escape(data));
          data = JSON.parse(jsonClean);
          return data;
        },
        responseType: "json",
      });

      return response.data;
```

**Comment by barbarbar338** [comment]
well i tried you solution but output is same
also escape is a deprecated method and i can't find anything that does the same thing

![preview](https://cdn.338.rocks/v1/storage/uploads/images/327112865423163512.png)
![preview](https://cdn.338.rocks/v1/storage/uploads/images/327112761706414199.png)

**Comment by nhonhoem0166** [comment]
it works for me
![image](https://user-images.githubusercontent.com/56016610/182271456-5f25b9c0-c816-4c61-b264-996985f19470.png)

**Comment by nhonhoem0166** [comment]
idk but it work for me
![image](https://user-images.githubusercontent.com/56016610/182272036-4513816a-c066-4e1f-91bf-2c4856730bef.png)

**Comment by nhonhoem0166** [comment]
> well i tried you solution but output is same also escape is a deprecated method and i can't find anything that does the same thing
> 
> ![preview](https://camo.githubusercontent.com/bbe0463bc70c829cc4a361bfc4f4dec93b6b6cc1c3f389f1c67e3e82454a5c78/68747470733a2f2f63646e2e3333382e726f636b732f76312f73746f726167652f75706c6f6164732f696d616765732f3332373131323836353432333136333531322e706e67) ![preview](https://camo.githubusercontent.com/5fccec7f33221a1c83dee9262aaed71855ecd26c4efda5fcc612059b01c57c55/68747470733a2f2f63646e2e3333382e726f636b732f76312f73746f726167652f75706c6f6164732f696d616765732f3332373131323736313730363431343139392e706e67)

```
function myEscape(str){
  var hex, i;

  var result = "";
  for (i=0; i<str.length; i++) {
      hex = str.charCodeAt(i).toString(16);
      result += ("%"+hex).slice(-4);
  }

  return result
}
```

**Comment by Rot4tion** [comment]
```
function myEscape(str) {
    return str.replaceAll("\\u00","%");
}
  
```

**Comment by Rot4tion** [comment]
you need to set try catch as follows to avoid unexpected errors

```
const response = await axios.get(url, {
        transformResponse: (data) => {
          try{
            data = decodeURIComponent(this.Escape(data));
          }
          catch{}
          return JSON.parse(data);
        },
        responseType: "json",
      });

      return response.data;
```

**Comment by barbarbar338** [comment]
yeah looks great now!

![preview](https://cdn.338.rocks/v1/storage/uploads/images/327374661006524542.png)

**Comment by barbarbar338** [comment]
if you want you can create a PR or i'll make a commit about this

**Comment by barbarbar338** [comment]
fixed with #4, thanks to @nhonhoem0166
